sekoia_automation/trigger.py

```diff
@@ -246,3 +256,52 @@ def run(self) -> None:
 
         Should usually be an infinite loop, calling send_event when relevant.
         """
+
+    def start_monitoring(self):
+        port = (
+            self.module.load_config(self.LIVENESS_PORT_FILE_NAME, non_exist_ok=True)
+            or 8000
+        )
+        LivenessHandler.trigger = self
+        self._liveness_server = HTTPServer(("", int(port)), LivenessHandler)
+        Thread(target=self._liveness_server.serve_forever, daemon=True).start()
+
+    def stop_monitoring(self):
+        if self._liveness_server:
+            self._liveness_server.shutdown()
+
+    def is_alive(self) -> bool:
+        """
+        Return whether the trigger appears to be alive.
+
+        This is based on the date of the last sent events
+        compared to the `seconds_without_events` threshold.
+        """
+        if (
+            self.seconds_without_events <= 0
+            or datetime.utcnow() - self._last_events
+            < timedelta(seconds=self.seconds_without_events)
+        ):
+            return True
+
+        delta = (datetime.utcnow() - self._last_events).seconds
```

Comment:
useless improvment : you could compute only once the delta and use it in both cases